x-pack/platform/test/alerting_api_integration/spaces_only/tests/alerting/group4/muted_alerts.ts

```diff
@@ -117,5 +144,247 @@ export default function createDisableRuleTests({ getService }: FtrProviderContex
       const mutedRule = rules.data.find((rule: { id: string }) => rule.id === createdRule1);
       expect(mutedRule.muted_alert_ids).to.contain(alertFromRule1._source[ALERT_INSTANCE_ID]);
     });
+
+    it('should set ALERT_MUTED field to true when muting an alert instance', async () => {
+      const ruleId = await createRule();
+
+      // Wait for alerts to be created
+      let alerts: any[] = [];
+      await retry.try(async () => {
+        alerts = await getAlertsByRuleId(ruleId);
+        expect(alerts.length).greaterThan(0);
+      });
+
+      const alertInstanceId = alerts[0]._source[ALERT_INSTANCE_ID];
+
+      expect(alerts[0]._source[ALERT_MUTED]).to.be(false);
+
+      // Mute the alert instance
+      await supertest
+        .post(
+          `${getUrlPrefix(Spaces.space1.id)}/api/alerting/rule/${encodeURIComponent(
+            ruleId
+          )}/alert/${encodeURIComponent(alertInstanceId)}/_mute`
+        )
+        .set('kbn-xsrf', 'foo')
+        .expect(204);
+
+      // Run the rule to trigger reconciliation
+      await runRuleNow(ruleId);
```

Comment:
As we update the alert docs when the API is called, do we need the rule to run? Or is it because we need more alerts to generated so we can also test the rule runner logic?

Reply:
We need `runRuleNow()` because the SO is the source of truth, and rule execution performs reconciliation. While the API call triggers `updateByQuery` with `wait_for_completion: false`, there's a risk that updates may fail or be incomplete. So this test ensures alert documents are synced with the saved object state when the rule runs